hibernate-core/src/test/java/org/hibernate/query/hhh14154/HHH14154Test.java

```diff
@@ -0,0 +1,60 @@
+package org.hibernate.query.hhh14154;
+
+import java.util.Date;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Test;
+
+import static org.hibernate.testing.transaction.TransactionUtil.doInJPA;
+
+/**
+ * @author Nathan Xu
+ */
+@TestForIssue( jiraKey = "HHH-14154" )
+public class HHH14154Test extends BaseCoreFunctionalTestCase {
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { HHH14154Test.Foo.class };
+	}
+
+	@Test
+	public void testNoExceptionThrown() {
+		doInJPA( this::sessionFactory, em -> {
+			final CriteriaBuilder cb = em.getCriteriaBuilder();
+			final CriteriaQuery<Foo> cq = cb.createQuery( Foo.class );
+			final Root<Foo> foo = cq.from( Foo.class );
+			cq.select(foo)
+				.where(
+					cb.lessThanOrEqualTo(
+						cb.concat(
+							cb.function( "FORMATDATETIME", String.class, foo.get( "startTime" ), cb.literal( "HH:mm:ss" ) ),
```

Comment:
I don't think every DBMS supports this function, so please add a `@RequiresDialect(H2Dialect.class)`

Reply:
done. thanks. also added the bug reporter as one of the authors.